src/hotspot/share/oops/typeArrayKlass.cpp

```diff
@@ -228,7 +228,6 @@ Klass* TypeArrayKlass::array_klass_or_null() {
 }
 
 size_t TypeArrayKlass::oop_size(oop obj) const {
-  assert(obj->is_typeArray(),"must be a type array");
```

Comment:
So, these checks are removed because they reach for class, which we cannot use with `UseCompactObjectHeaders`? Better to disable the assert at this time, I think, instead of removing it.